### PR TITLE
feat(vm): support wasm api sendTx/readTx

### PR DESCRIPTION
**Pull request by huangzhiran**

### CODE REVIEW

File: pkg/modules/vm/wasmapi/async/processor.go [hunk 1]
```diff
@@ -47,12 +48,15 @@ func (p *ApiCallProcessor) ProcessTask(ctx context.Context, t *asynq.Task) error
 		return fmt.Errorf("http.ReadRequest failed: %v: %w", err, asynq.SkipRetry)
 	}
 
+	req.Header.Set(apitypes.W3bstreamSystemProjectID, payload.Project.ProjectID.String())
```
Comment (CoderZhi) [inline pkg/modules/vm/wasmapi/async/processor.go:51]: shall we use [func NewRequestWithContext](https://pkg.go.dev/net/http#NewRequestWithContext) to include `ctx`?

File: pkg/modules/vm/wasmapi/types/types.go [hunk 2]
```diff
@@ -5,6 +5,8 @@ import (
 	"net/http"
 )
 
+const W3bstreamSystemProjectID = "W3bstreamSystemProjectID"
```
Comment (Liuhaai) [inline pkg/modules/vm/wasmapi/types/types.go:8]: what's this for?

File: pkg/modules/vm/wasmapi/async/task.go [hunk 3]
```diff
@@ -12,14 +14,14 @@ const (
 )
 
 type apiCallPayload struct {
-	ProjectName string
-	Data        []byte
+	Project *models.Project
```
Comment (Liuhaai) [inline pkg/modules/vm/wasmapi/async/task.go:17]: payload in the task should be binary. Add `Serialize() []byte` and `Deserialize([]byte) *apiCallPayload` for apiCallPayload clas

File: cmd/srv-applet-mgr/global/config.go [hunk 4]
```diff
@@ -139,7 +139,7 @@ func init() {
 
 	redisKvDB := kvdb.NewRedisDB(config.Redis)
 
-	wasmApiServer, err := wasmapi.NewServer(std, config.Redis, config.Postgres, redisKvDB)
+	wasmApiServer, err := wasmapi.NewServer(std, config.Redis, config.Postgres, redisKvDB, config.EthClient)
```
Comment (Liuhaai) [inline cmd/srv-applet-mgr/global/config.go:142]: build the handlers outside of server init, and pass them to the server

File: pkg/modules/vm/wasmapi/types/types.go [hunk 5]
```diff
@@ -5,6 +5,8 @@ import (
...
+
```
Comment (Liuhaai) [inline pkg/modules/vm/wasmapi/types/types.go:9]: move to `NewRequest`?

File: pkg/modules/vm/wasmapi/handler/hello.go [hunk 6]
```diff
@@ -1,12 +1,12 @@
-package wasmapi
+package handler
 
 import (
 	"fmt"
 
 	"github.com/gin-gonic/gin"
 )
```
Comment (Liuhaai) [inline pkg/modules/vm/wasmapi/handler/hello.go:8]: type Handler interface{
    Handle()
}
type helloHanler {}

func NewHelloHandler() Handler {}

func (*helloHanler) Handle(){}

File: cmd/srv-applet-mgr/global/config.go (hunk 4 above)
Comment (huangzhiran) [reply cmd/srv-applet-mgr/global/config.go:142]: handlers within the wasmapi package would be better, outside woun't use it

**Review by hunshenshi — APPROVED** [review]

**Pull request merged**